src/awkward/_v2/_util.py

```diff
@@ -1316,204 +1267,118 @@ def wrap(content, behavior=None, highlevel=True, like=None):
-#         return ak._v2.contents.ListOffsetArrayU32(
-#             layout.offsets,
-#             transform(layout.content, depth + 1, user),
-#             layout.identities,
-#             layout.parameters if keep_parameters else None,
-#         )
-
-#     elif isinstance(layout, ak._v2.contents.ListOffsetArray64):
-#         return ak._v2.contents.ListOffsetArray64(
-#             layout.offsets,
-#             transform(layout.content, depth + 1, user),
-#             layout.identities,
-#             layout.parameters if keep_parameters else None,
-#         )
-
-#     elif isinstance(layout, ak._v2.contents.IndexedArray32):
-#         return ak._v2.contents.IndexedArray32(
-#             layout.index,
-#             transform(layout.content, depth, user),
-#             layout.identities,
-#             layout.parameters if keep_parameters else None,
-#         )
-
-#     elif isinstance(layout, ak._v2.contents.IndexedArrayU32):
-#         return ak._v2.contents.IndexedArrayU32(
-#             layout.index,
-#             transform(layout.content, depth, user),
-#             layout.identities,
-#             layout.parameters if keep_parameters else None,
-#         )
-
-#     elif isinstance(layout, ak._v2.contents.IndexedArray64):
-#         return ak._v2.contents.IndexedArray64(
-#             layout.index,
-#             transform(layout.content, depth, user),
-#             layout.identities,
-#             layout.parameters if keep_parameters else None,
-#         )
-
-#     elif isinstance(layout, ak._v2.contents.IndexedOptionArray32):
-#         return ak._v2.contents.IndexedOptionArray32(
-#             layout.index,
-#             transform(layout.content, depth, user),
-#             layout.identities,
-#             layout.parameters if keep_parameters else None,
-#         )
-
-#     elif isinstance(layout, ak._v2.contents.IndexedOptionArray64):
-#         return ak._v2.contents.IndexedOptionArray64(
-#             layout.index,
-#             transform(layout.content, depth, user),
-#             layout.identities,
-#             layout.parameters if keep_parameters else None,
-#         )
-
-#     elif isinstance(layout, ak._v2.contents.ByteMaskedArray):
-#         return ak._v2.contents.ByteMaskedArray(
-#             layout.mask,
-#             transform(layout.content, depth, user),
-#             layout.valid_when,
-#             layout.identities,
-#             layout.parameters if keep_parameters else None,
-#         )
-
-#     elif isinstance(layout, ak._v2.contents.BitMaskedArray):
-#         return ak._v2.contents.BitMaskedArray(
-#             layout.mask,
-#             transform(layout.content, depth, user),
-#             layout.valid_when,
-#             len(layout),
-#             layout.lsb_order,
-#             layout.identities,
-#             layout.parameters if keep_parameters else None,
-#         )
+def transform_child_layouts(transform, layout, depth, user=None, keep_parameters=True):
```

Comment:
The `transform_child_layouts` was once part of `recursively_apply`, which is now a method on Content. It should be possible to express what you're using `transform_child_layouts` for with `recursively_apply` (and thus reduce the number of code paths to support—that's why they were consolidated).